### PR TITLE
Likely fix for cross-z tts and misc sendspeech improvements

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request
humans_by_zlevel works now and replace another one i added
TTS should no longer cross z for ghosts, tm would be good

## Changelog
:cl:
fix: fixed  TTS for ghosts going across zlevels
/:cl:

### CODE REVIEW

**Comment by Strayed-Raven** [comment]
I told that you can hear TTS in certain spots as Ghost even outside their range. Knew it

File: code/game/sound.dm [hunk 1]
```diff
@@ -143,10 +143,10 @@ A good representation is: 'byond applies a volume reduction to the sound every X
 ///Play a sound for all cliented humans and ghosts by zlevel
 /proc/playsound_z_humans(z, soundin, _volume)
 	soundin = sound(get_sfx(soundin), channel = SSsounds.random_available_channel(), volume = _volume)
-	for(var/mob/living/carbon/human/H AS in GLOB.humans_by_zlevel["[z]"])
+	for(var/mob/living/carbon/human/H AS in SSmobs.dead_players_by_zlevel[z])
```
Comment (Lumipharon) [inline code/game/sound.dm:146]: ??

**Review by Lumipharon — CHANGES_REQUESTED** [review]

**Comment by Texicos** [comment]
broke all groundside local chat on lawanka

**Comment by Lumipharon** [comment]
> broke all groundside local chat on lawanka

I don't see this?

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**